router/http.go

```diff
@@ -166,15 +166,22 @@ func (s *HTTPListener) doSync(ctx context.Context, errc chan<- error) <-chan str
 
 func (s *HTTPListener) startListen() error {
 	if err := s.listenAndServe(); err != nil {
+		s.Close()
 		return err
 	}
-	s.Addr = s.listener.Addr().String()
+	s.Addrs = []string{}
```

Comment:
Done. Is that just for efficiency, or is there a deeper reason to do it that way?